### PR TITLE
Feature/unit tests

### PR DESCRIPTION
**Pull request by mkryva**

### CODE REVIEW

File: proxysql_tools/managers/proxysql_manager.py [hunk 1]
```diff
@@ -28,6 +28,7 @@ def __init__(self, host, port, user, password, socket=None,
         :param reload_runtime: Whether the ProxySQL runtime should be
             reloaded for the changes to take affect.
         :type reload_runtime: bool
+        :raise: ProxySQLAdminConnectionError
```
Comment (ovaistariq) [inline proxysql_tools/managers/proxysql_manager.py]: `__init__` does not raise this exception. Perhaps you want to add it to the docstring of `ping` function.

File: tests/unit/test_proxysql_manager.py [hunk 2]
```diff
@@ -53,3 +54,103 @@ def test__deregister_backend_does_not_delete_the_backend_if_not_registered():
...
+        .and_return(True))
+
+    (expect(proxysql_man)
+        .insert_or_update_mysql_backend
+        .once())
+
+    proxysql_man.update_mysql_backend_status(backend.hostgroup_id, backend.hostname, backend.port, "dummy_status")
+
+
+def test_update_mysql_backend_status_if_backend_not_registered():
```
Comment (ovaistariq) [inline tests/unit/test_proxysql_manager.py]: Unit test function names should be named after the action which we want to test.
For example a better name would be:
`test__update_mysql_backend_status_not_called_if_backend_not_registered`

File: tests/unit/test_proxysql_manager.py [hunk 3]
```diff
@@ -53,3 +54,103 @@ def test__deregister_backend_does_not_delete_the_backend_if_not_registered():
 
     proxysql_man.deregister_backend(backend.hostgroup_id, backend.hostname,
                                     backend.port)
+
+
+def test_register_backend():
```
Comment (ovaistariq) [inline tests/unit/test_proxysql_manager.py]: Existing function names follow the convention of having the function name start with `test__` (two underscores), please use the same convention.

File: tests/unit/test_proxysql_manager.py [hunk 4]
```diff
@@ -53,3 +54,103 @@ def test__deregister_backend_does_not_delete_the_backend_if_not_registered():
...
+
+    (expect(proxysql_man)
+        .insert_or_update_mysql_backend
+        .never())
+
+    with pytest.raises(ProxySQLMySQLBackendUnregistered):
+        proxysql_man.update_mysql_backend_status(backend.hostgroup_id, backend.hostname, backend.port, "dummy_status")
+
+
+def test_register_mysql_user_if_registered():
```
Comment (ovaistariq) [inline tests/unit/test_proxysql_manager.py]: Rename the function to `test__register_mysql_user_does_not_update_if_already_registered`

File: proxysql_tools/managers/proxysql_manager.py (hunk 1 above)
Comment (mkryva) [inline proxysql_tools/managers/proxysql_manager.py]: Ah, really. Yes, You're right.

File: tests/unit/test_proxysql_manager.py [hunk 5]
```diff
@@ -53,3 +54,103 @@ def test__deregister_backend_does_not_delete_the_backend_if_not_registered():
...
+        .and_return(MagicMock()))
+
+    (expect(proxysql_man)
+        .insert_or_update_mysql_backend
+        .once())
+
+    proxysql_man.register_backend(backend.hostgroup_id, backend.hostname, backend.port)
+
+
+def test__update_mysql_backend_status_not_called_if_backend_registered():
```
Comment (ovaistariq) [inline tests/unit/test_proxysql_manager.py]: Did you mean `test__update_mysql_backend_status_called_if_backend_registered` ?

**Comment by ovaistariq** [comment]
Looks good, so I am going to merge it!

**Pull request merged**